marklogic-data-hub/src/main/java/com/marklogic/hub/provenance/ProvenanceManager.java

```diff
@@ -12,7 +17,28 @@ public ProvenanceManager(HubClient hubClient) {
     }
 
     public void deleteProvenanceRecords(String retainDuration) {
+        deleteProvenanceRecords(retainDuration, DatabaseKind.JOB.name());
+    }
+
+    public void deleteProvenanceRecords(String retainDuration, String database) {
         String apiPath = "ml-modules/root/data-hub/5/data-services/provenance/deleteProvenance.api";
-        BulkUtil.deleteData(hubClient.getJobsClient(), apiPath, retainDuration);
+        DatabaseKind databaseKind = DatabaseKind.valueOf(database);
+        DatabaseClient client = null;
+        switch (databaseKind) {
+            case JOB: client = hubClient.getJobsClient();
+                break;
+            case STAGING: client = hubClient.getStagingClient();
+                break;
+            case FINAL: client = hubClient.getFinalClient();
+                break;
+            default: throw new RuntimeException("Invalid provenance database specified!");
```

Comment:
Should include the "database" value in here - e.g. "Invalid provenance database: " + database